### PR TITLE
Fix race of create tag and edge

### LINKED ISSUE #3376: Concurrent create tag and edge with same name may be both succeed.
**Please check the FAQ documentation before raising an issue**

Please check the [FAQ](https://github.com/vesoft-inc/nebula/blob/master/docs/manual-EN/1.overview/2.quick-start/2.FAQ.md) documentation and old issues before raising an issue in case someone has asked the same question that you are asking.

**Describe the bug (__required__)**

```cpp
 // CreateTagProcessor.cpp
  {
    // if there is an edge of the same name
    // TODO: there exists race condition, we should address it in the future
    folly::SharedMutex::ReadHolder rHolder(LockUtils::edgeLock());
    auto conflictRet = getEdgeType(spaceId, tagName);
    ......
  }  // <------------------------Unlock!!!!!!!!!
  auto columns = req.get_schema().get_columns();
  ......
```
Because `edgeLock` unlock before finish create tag, if there are two threads create tag/edge concurrency, may be both succeed.

**Your Environments (__required__)**

* OS: `uname -a`
* Compiler: `g++ --version` or `clang++ --version`
* CPU: `lscpu`
* Commit id (e.g. `a3ffc7d8`)

**How To Reproduce(__required__)**

1. Step 1
2. Step 2
3. Step 3

**Expected behavior**

**Additional context**

Provide logs and configs, or any other context to trace the problem.

### PR DESCRIPTION
**Pull request by jackwener**
#### What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### What problem(s) does this PR solve?
Issue(s) number:  close #3376

Description:

Merge two lock

#### Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by jackwener** [comment]
why not increase the scope of lock.
Because now
```
(Edge RL) --> (Tag WL)
```
if 
```
(Edge RL --> (Tag WL)  ）
```
This will be dead lock

File: src/meta/processors/schema/CreateEdgeProcessor.cpp [hunk 1]
```diff
@@ -13,28 +13,26 @@ namespace meta {
...
-    // TODO: there exists race condition, we should address it in the future
-    folly::SharedMutex::ReadHolder rHolder(LockUtils::tagLock());
-    auto conflictRet = getTagId(spaceId, edgeName);
-    if (nebula::ok(conflictRet)) {
-      LOG(ERROR) << "Failed to create edge `" << edgeName
-                 << "': some tag with the same name already exists.";
-      resp_.id_ref() = to(nebula::value(conflictRet), EntryType::EDGE);
-      handleErrorCode(nebula::cpp2::ErrorCode::E_CONFLICT);
+  const auto& edgeName = req.get_edge_name();
+  folly::SharedMutex::ReadHolder rHolder(LockUtils::tagAndEdgeLock());
```
Comment (Shylock-Hg) [inline src/meta/processors/schema/CreateEdgeProcessor.cpp:17]: WriteHolder.

File: src/meta/processors/schema/CreateTagProcessor.cpp [hunk 2]
```diff
@@ -13,28 +13,26 @@ namespace meta {
...
-    // TODO: there exists race condition, we should address it in the future
-    folly::SharedMutex::ReadHolder rHolder(LockUtils::edgeLock());
-    auto conflictRet = getEdgeType(spaceId, tagName);
-    if (nebula::ok(conflictRet)) {
-      LOG(ERROR) << "Failed to create tag `" << tagName
-                 << "': some edge with the same name already exists.";
-      resp_.id_ref() = to(nebula::value(conflictRet), EntryType::TAG);
-      handleErrorCode(nebula::cpp2::ErrorCode::E_CONFLICT);
+  const auto& tagName = req.get_tag_name();
+  folly::SharedMutex::ReadHolder rHolder(LockUtils::tagAndEdgeLock());
```
Comment (Shylock-Hg) [inline src/meta/processors/schema/CreateTagProcessor.cpp:17]: ditto.

Comment (jackwener) [reply src/meta/processors/schema/CreateTagProcessor.cpp]: fixed

**Review by Shylock-Hg — APPROVED** [review]

**Review by critical27 — APPROVED** [review]
LGTM, good job. 

BTW, we could unify all locks in meta with only one lock later, several reasons:
1. Dead lock may happen
2. Many interfaces does not use right lock

Contact me if you're interested

File: src/meta/processors/Common.h [hunk 3]
```diff
@@ -25,8 +25,7 @@ class LockUtils {
   GENERATE_LOCK(id);
   GENERATE_LOCK(workerId);
   GENERATE_LOCK(localId);
-  GENERATE_LOCK(tag);
-  GENERATE_LOCK(edge);
+  GENERATE_LOCK(tagAndEdge);
   GENERATE_LOCK(tagIndex);
   GENERATE_LOCK(edgeIndex);
```
Comment (critical27) [inline src/meta/processors/Common.h:30]: You could unify tagIndex and edgeIndex as one. Although tag/edge index is hard to misused, but they do will have same name IIRC.

**Pull request merged**